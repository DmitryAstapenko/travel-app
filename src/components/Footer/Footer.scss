.footer {
  padding: 0.5rem 0;
  display: flex;
  justify-content: space-between;

  & .authors {
    margin: auto 0;
  }

  & .logo {
    width: 100px;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
  }

  & .year {
    display: flex;
    align-items: flex-end;
    margin-right: 20px;
    margin-bottom: 10px;    
  }
}

@media (max-width: 991px) {
  .footer {
    font-size: 14px;
    
    & .nav {
      flex-direction: column;
    }
    
    & .nav-link {
      padding: 0;
    }

    & .year {
      margin-bottom: 0;      
    }
  }
}
