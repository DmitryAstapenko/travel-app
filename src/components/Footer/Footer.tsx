import * as React from "react";
import { Container, Nav } from "react-bootstrap";
import LOCALIZATIONS from "../../assets/data/localizations";
import LocalizationContext from "../../context/LocalizationContext";
import "./Footer.scss";

const Footer: React.FunctionComponent = () => {
  const language = React.useContext(LocalizationContext);
  return (
    <Container>
      <div className="footer">
        <Nav className="mr-auto">
          <h5 className="authors">{`${LOCALIZATIONS.authors[language]}:`}</h5>
          &nbsp; &nbsp;
          <Nav.Link href="https://github.com/DmitryAstapenko">
            DmitryAstapenko
          </Nav.Link>
          <Nav.Link href="https://github.com/RamanNovikau">
            RamanNovikau
          </Nav.Link>
          <Nav.Link href="https://github.com/naTpuk-ep">
            YauheniPiatrou
          </Nav.Link>
          <Nav.Link href="https://github.com/zhenyamarinovich">
            ZhenyaMarinovich
          </Nav.Link>
        </Nav>
        <div className="logo">
          <a href="https://rs.school/js/">
            <img
              src="https://rs.school/images/rs_school_js.svg"
              alt="RS-school"
            />
          </a>
        </div>
      </div>
    </Container>
  );
};

export default Footer;
