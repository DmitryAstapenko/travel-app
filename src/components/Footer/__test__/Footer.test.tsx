import React from "react";
import Footer from "../Footer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const setUp = () => shallow(<Footer />);

describe("should render footer component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it("should contain footer-container", () => {
    const wrapper = component.find(".footer");
    expect(wrapper.length).toBeLessThan(2);
  });

  it("should render footer component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
