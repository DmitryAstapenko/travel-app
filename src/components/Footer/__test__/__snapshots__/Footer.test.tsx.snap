// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render footer component should render footer component 1`] = `
<Container
  fluid={false}
>
  <div
    className="footer"
  >
    Footer
  </div>
</Container>
`;
